test/e2e_node/resource_metrics_test.go

```diff
@@ -41,36 +40,24 @@ const (
 	maxStatsAge = time.Minute
 )
 
-var _ = framework.KubeDescribe("ResourceMetricsAPI", func() {
+var _ = framework.KubeDescribe("ResourceMetricsAPI [NodeFeature:ResourceMetrics]", func() {
 	f := framework.NewDefaultFramework("resource-metrics")
 	ginkgo.Context("when querying /resource/metrics", func() {
 		ginkgo.BeforeEach(func() {
-			ginkgo.By("Creating test pods")
+			ginkgo.By("Creating test pods to measure their resource usage")
 			numRestarts := int32(1)
 			pods := getSummaryTestPods(f, numRestarts, pod0, pod1)
 			f.PodClient().CreateBatch(pods)
-
-			ginkgo.By("Waiting for test pods to restart the desired number of times")
-			gomega.Eventually(func() error {
-				for _, pod := range pods {
-					err := verifyPodRestartCount(f, pod.Name, len(pod.Spec.Containers), numRestarts)
```

Comment:
This was initially added to catch a bug in which container metrics would disappear after container restarts.  It was copied from the summary API test IIRC.  I'd like to keep this if possible.

Reply:
I put back the loop with your explanation as comment.
I'm not sure this is going to have the desired effect since we wait later for the metrics to populate. WDYT?

Reply:
Right.  IIRC, the bug that we fixed, and added this test for was where we only collected metrics for the first instance of a container, and not for ones created after the container restarts.  So we want to first have a pod with containers that have already restarted, and then make sure metrics look correct after that.  Make sense?

Reply:
I understand now. I'll think on this and if it needs to be rewritten again. Thank you.

Reply:
I think this should be 
BY "restarting containers to ensure that future instances of a container besides the first instance collect metrics"
or something like that. 

Is there an issue/PR we can link to for demonstative purposes? Or does this need a separate Test case itself?